### PR TITLE
New: Add promotion of free add-on for free users

### PR DESCRIPTION
**Pull request by JasonTheAdams**
## Description

This adds promotional offerings for a free add-on. This includes:
- A Free Add-On tab to the add-ons page
- A modal that pops up upon updating GiveWP (and the next two versions if dismissed)
- A green button that links to the add-on page on all Give Settings pages

## Testing Instructions

1. Use the different forms to subscribe
2. Ask someone with Active Campaign access to confirm new user and tags

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@ravinderk What is the best way to check if a customer has _any_ active premium license?

**Comment by ravinderk** [comment]
> @ravinderk What is the best way to check if a customer has _any_ active premium license?

@JasonTheAdams you can use this function from `PremiumAddonsListManager` class:
https://github.com/impress-org/givewp/blob/0251861623cb40c835340b60b2c7a2ddc2b67b87/src/License/PremiumAddonsListManager.php#L85

**Comment by JasonTheAdams** [comment]
@ravinderk I don't think you understood my question. Basically, I have something that I want to show only if no premium license is active (either a Plan license or single add-on). Does that make sense?

**Comment by ravinderk** [comment]
> @ravinderk I don't think you understood my question. Basically, I have something that I want to show only if no premium license is active (either a Plan license or a single add-on). Does that make sense?

@JasonTheAdams I think, GiveWP core does not have anything built-in.  You can create a function to check whether only a single license or plan is active. To do that you have to fetch all license from `get_option( 'give_licenses', [] )` and filter them.
If the license has `is_all_access_pass` then it means it is a bundle plan otherwise single.

For example, you can check how I am rendering licenses ( single and bundle) on the admin license page.
https://github.com/impress-org/givewp/blob/0f95d8c75bcc95c05ea96e4ef208cf01969bc0a9/includes/class-give-license-handler.php#L535

File: src/Promotions/FreeAddonModal/Controllers/DisplaySettingsButton.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+     * @return void
+     */
+    public function __invoke()
+    {
+        if ( !$this->displayOffer() ) {
+            return;
+        }
+
+        $url = admin_url( 'edit.php?post_type=give_forms&page=give-add-ons&tab=3' );
+        echo "<a class='give-green-button i-crown' href='$url' target='_blank'>Get a Free Add-On</a>";
```
Comment (ravinderk) [inline src/Promotions/FreeAddonModal/Controllers/DisplaySettingsButton.php:19]: @JasonTheAdams can we make this string translatable?

File: src/Promotions/FreeAddonModal/resources/Modal.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    const handleDismiss = (event) => {
+        event.preventDefault();
+        setIsOpen(false);
+        rejectOffer();
+    };
+
+    return (
+        <div className={styles.giveModalContainer}>
+            <div className={styles.modal}>
+                <h2>🎉 Congratulations!</h2>
```
Comment (ravinderk) [inline src/Promotions/FreeAddonModal/resources/Modal.js:34]: @JasonTheAdams do we make these strings translatable?

File: src/Promotions/InPluginUpsells/resources/js/components/FreeAddOnTab.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+import styles from './FreeAddOnTab.module.css';
+import {Hero} from './Hero';
+import {Card} from './Card';
+import {transformStrong} from '../utils';
+
+const highlights = [
+    {
+        icon: 'https://givewp.com/downloads/upsells/images/time-icon.svg',
+        text: 'Flexible Delivery Times',
```
Comment (ravinderk) [inline src/Promotions/InPluginUpsells/resources/js/components/FreeAddOnTab.js:15]: @JasonTheAdams can we make these strings translatable?

File: src/Promotions/InPluginUpsells/resources/js/components/FreeAddOnTab.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            </Card>
+            <Hero heading={form.heading} className={styles.informationHero} />
+            <Card className={cx(styles.card, styles.formCard)}>
+                {userSubscribed ? (
+                    <span className={styles.formCard__thankYou}>{form.submissionConfirmation}</span>
+                ) : (
+                    <>
+                        {hasSubscriptionError && (
+                            <div className={styles.submissionError}>
+                                There was an issue submitting your information. Please try again. If the problem
```
Comment (ravinderk) [inline src/Promotions/InPluginUpsells/resources/js/components/FreeAddOnTab.js:76]: @JasonTheAdams can we make these strings translatable?

**Comment by JasonTheAdams** [comment]
@mathetos @DevinWalker I'm curious of your thoughts regarding @ravinderk's comments above about making things translatable. Normally, I'm all about it. But in this case it's a bit different for two reasons:
1. This is purely promotional. It's not a feature of GiveWP, it's just a promotion. So how eager will folks be to translate it?
2. Much of this leads back to the add-ons page, and that's straight up not translatable. We'd have to translate it ourselves and provide unique translation files per language, such as `addons.es.json`. Therefore, unless we go about doing translations ourselves for that, only part of this flow will ever truly be translatable.

What do you think?

File: src/Promotions/InPluginUpsells/resources/js/components/FreeAddOnTab.js (hunk 3 above)
Comment (JasonTheAdams) [reply src/Promotions/InPluginUpsells/resources/js/components/FreeAddOnTab.js]: This is just leftover mock data. I've removed it.

File: src/Promotions/InPluginUpsells/resources/js/components/FreeAddOnTab.js (hunk 4 above)
Comment (JasonTheAdams) [reply src/Promotions/InPluginUpsells/resources/js/components/FreeAddOnTab.js:61]: There's not really any point. Literally every other string on this page is only ever going to be in English, unless we start providing our own translations on our server.

**Comment by DevinWalker** [comment]
No need to translate at this time...

**Review by DevinWalker — APPROVED** [review]

**Pull request merged**